Add notes on any source and any port

susam opened the pull request:
See https://github.com/cloudmarker/guides/blob/any-source-any-port/tech/firewall-rule-format.md for the rendered README.

Here are the direct links to the rendered new sections:

- [Any Source](https://github.com/cloudmarker/guides/blob/any-source-any-port/tech/firewall-rule-format.md#any-source)
- [Any Port](https://github.com/cloudmarker/guides/blob/any-source-any-port/tech/firewall-rule-format.md#any-port)

s-nirali commented on tech/firewall-rule-format.md:
```diff
@@ -489,7 +514,7 @@ potentially weak:
     if they are allowed to all source address ranges.
```
I missed this in my earlier review. In the [Multiple Ranges](https://github.com/cloudmarker/guides/blob/any-source-any-port/tech/firewall-rule-format.md#multiple-ranges) section, the sentence "There are primarily three important pieces of information..." should be "There are primarily **four** important pieces of information...".

susam replied:
Thank you for pointing this out. I will fix this in a separate pull request in order to not mix separate concerns in the same pull request. In fact, based on cloudmarker/cloudmarker#69 that I am working on, this section needs a few more updates too which I will take care of in a separate pull request.

s-nirali commented on tech/firewall-rule-format.md:
"GCP contains this firewall rule to allow connectivity **to** port 22 from"

susam replied:
Addressed in [line 781](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R781).

s-nirali commented on tech/firewall-rule-format.md:
I was just wondering, is the case for multiple CIDRs not applicable for GCP?

susam replied:
It is still applicable. The reason why we have `"source_address_prefixes": ["0.0.0.0/1", "128.0.0.0/1"]` in the Azure example but we only have `"sourceRanges": ["0.0.0.0/0"]` in the GCP example is that the Azure portal did not let me enter `0.0.0.0/0` as a valid CIDR at all, so just for the sake of writing an example I entered two CIDRs that when combined is equivalent to `0.0.0.0/0`.

Having said that, I am not planning to perform union of CIDRs to see if it is equivalent to the entire Internet in the initial implementation. That is a very interesting problem to solve though.

s-nirali commented on tech/firewall-rule-format.md:
`Destination port range` is "*". `Destination port ranges` is "[]".

susam replied:
The italicized text are labels I have picked up verbatim from Azure Portal UI. So here I mean that I entered `*` in the *Destination port ranges* in the Azure Portal UI.

<img width="585" alt="Screen Shot 2019-04-01 at 2 44 02 PM" src="https://user-images.githubusercontent.com/229488/55316772-bdeb9b80-548c-11e9-8ce2-798a97e78faf.png">

Azure decides to put the port range in either `destination_port_range` or `destination_port_ranges` in the API response depending on whether one or more than one port ranges are specified.

Does this make sense now?

I will clarify this notation convention somewhere at the top of the document and update this pull request.

susam replied:
Clarified the notation at [line 46](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R46).

s-nirali replied:
This makes sense. Thank you for the clarification.

s-nirali commented on tech/firewall-rule-format.md:
The key here is also `destination_port_range`.

susam replied:
See https://github.com/cloudmarker/guides/pull/5#discussion_r270779322.

s-nirali commented on tech/firewall-rule-format.md:
```diff
@@ -575,15 +600,487 @@ For the record, the reason why the protocol is `*` in Azure is that
+`Invalid argument: 'Prefix'. Reason: The prefix must be between 1 and 32.`.
+
+The weakest rule we could configure in Azure portal using a single CIDR
+is `0.0.0.0/1`. Whe an inbound security rule is configured with *IP
```
`When an..`

susam replied:
Fixed at [line 682](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R682).

s-nirali commented on tech/firewall-rule-format.md:
Single `we`.

susam replied:
Fixed at [line 962](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R962).

susam commented:
Addressed the review comments. The pull request is already in approved state. So I will go ahead and merge this. If there are more review comments, please feel free to share them. I will address them in a future pull request.
